### PR TITLE
Add Prometheus Remote Write Exporter supporting Cortex - Factory and Config

### PR DESCRIPTION
**Pull request by danielbang907**
**Description:** 
This PR implements the config, config YAML, and factory structs, the prwExporter struct. It also includes the README.md for a sample configuration to run this Prometheus remote-write exporter. The config and factory structs are fully compatible with the Collector component interfaces. 

**Link to tracking Issue:**
#1150
Related issues are: 
Metrics aggregation proposal: #1422
Prometheus exporter not functional: #1255
Related spec discussion: #731 

**Testing:** 
The unit tests coverage is currently at 86.2% for this pull request. However combining this PR with other related PRs increases the coverage to 92.1%.

**Documentation:**
* readme.md with sample configuration, file structure, and assumptions
* Design documentation: * *PR #1464

cc: @huyan0 @danielbang907 @alolita @markcartertm @sonofachamp @morigs 
Request for review: @jmacd @bogdandrutu @annanay25 @gouthamve @open-telemetry/collector-approvers  @open-telemetry/collector-maintainers

### CODE REVIEW

**Comment by bogdandrutu** [comment]
Please fix lint errors

**Comment by danielbang907** [comment]
Hi @bogdandrutu, will do! I will also add that the PR #1525 actually contains the total codebase (everything in this PR + logic for the exporter) Yang and I worked on, so there is no need to pull this codebase, should the other PR gets merged.

**Comment by bogdandrutu** [comment]
@danielbang907 would prefer to merge this separated, easier to review

File: exporter/prometheusremotewriteexporter/factory.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+)
+
+func NewFactory() component.ExporterFactory {
+	return exporterhelper.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		exporterhelper.WithMetrics(createMetricsExporter))
+}
+
+// Instantiates a pseudo-Cortex Exporter that adheres to the component MetricsExporter interface
```
Comment (bogdandrutu) [inline exporter/prometheusremotewriteexporter/factory.go:41]: Comments in golang should start with the func name:
"// createMetricsExporter instantiates ... "

File: exporter/prometheusremotewriteexporter/factory.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	return prwexp, nil
+}
+
+// Instantiates the default version of a Remote Write Exporter config struct to
+// be used by the factory to build Exporter instances
+func createDefaultConfig() configmodels.Exporter {
```
Comment (bogdandrutu) [inline exporter/prometheusremotewriteexporter/factory.go:77]: Same for func names

File: exporter/prometheusremotewriteexporter/factory_test.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configcheck"
+	"go.opentelemetry.io/collector/config/configmodels"
+)
+
+//Tests whether or not the default Exporter factory can instantiate a properly interfaced Exporter with default conditions
+func TestCreateDefaultConfig(t *testing.T) {
+	factory := NewFactory()
+	cfg := factory.CreateDefaultConfig()
```
Comment (bogdandrutu) [inline exporter/prometheusremotewriteexporter/factory_test.go:31]: can call directly `createDefaultConfig`

File: exporter/prometheusremotewriteexporter/factory_test.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	assert.NotNil(t, cfg, "failed to create default config")
+	assert.NoError(t, configcheck.ValidateConfig(cfg))
+}
+
+//Tests whether or not a correct Metrics Exporter from the default Config parameters
+func TestCreateMetricsExporter(t *testing.T) {
+	factory := NewFactory()
```
Comment (bogdandrutu) [inline exporter/prometheusremotewriteexporter/factory_test.go:38]: same no need to have a factory

File: exporter/prometheusremotewriteexporter/README.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+File structure:
+
+- `cortex.go`: exporter implementation. Converts and sends OTLP metrics
+
+- `helper.go`: helper functions that cortex.go uses. Performs tasks such as sanitizing label and generating signature string
+
+- `config.go`: configuration struct of the exporter
+
+- `factory.go`: initialization methods for creating default configuration and the exporter
```
Comment (bogdandrutu) [inline exporter/prometheusremotewriteexporter/README.md:48]: This will soon get outdated, recommend to not have it here.

**Comment by danielbang907** [comment]
Closing this PR as @huyan0 is filing one that addresses the current comments and CI checks.

**Pull request closed without merging**